util/pkg/distributions/distributions.go

```diff
@@ -96,17 +100,55 @@ func (d *Distribution) DefaultUsers() ([]string, error) {
 
 // HasLoopbackEtcResolvConf is true if systemd-resolved has put the loopback address 127.0.0.53 as a nameserver in /etc/resolv.conf
 // See https://github.com/coredns/coredns/blob/master/plugin/loop/README.md#troubleshooting-loops-in-kubernetes-clusters
-func (d *Distribution) HasLoopbackEtcResolvConf() bool {
-	switch d.project {
-	case "debian":
-		return d.version >= 12
-	case "ubuntu":
-		return d.version >= 18.04
-	case "flatcar":
-		return true
-	default:
-		return false
+// Returns the recommended target if we should not use /etc/resolv.conf
+//
+// There are actually 3 main configurations:
+//
+// * "Classic" DNS:
+//     upstream nameservers are in /etc/resolv.conf.
+//     We can use /etc/resolv.conf
+//
+// * systemd-resolved without libnss_resolve:
+//     /etc/resolv.conf is a symlink to /run/systemd/resolve/stub-resolv.conf
+//     which contains 127.0.0.53 (that won't work in a container), and upstream
+//     nameservers are in /run/systemd/resolve/resolv.conf.
+//     We must use /run/systemd/resolve/resolv.conf
+//
+// * systemd-resolved with libnss_resolve:
+//     /etc/resolv.conf has upstream nameservers. /etc/nsswitch.conf includes
+//     "resolve", which is a direct-path to systemd-resolved.
+//     We can use /etc/resolv.conf
+
+func (d *Distribution) HasLoopbackEtcResolvConf() (string, bool) {
```

Comment:
Do we still need the bool, could just return a string ptr or `nil`.
```suggestion
func (d *Distribution) HasLoopbackEtcResolvConf() *string {
```

Reply:
Seems like a minor nuisance.